### PR TITLE
Refactor server

### PR DESCRIPTION
**Pull request by Joabesv**
Mini refactor to make this code less buggy

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -1,22 +1,14 @@
-import fastify from 'fastify';
-import { config } from './config/env';
-import { prettyLog } from './utils/prettyLog';
+import { buildServer } from "@/server/";
 
-const server = fastify({
-  logger: {
-    transport: prettyLog,
```
Comment (oh-math) [inline src/server.ts:7]: One doubt... This piece is creating a log struct with a configuration that prettify logs?

File: src/server.ts [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
-});
-
-async function buildServer() {
+async function server() {
+  const app = await buildServer();
+  const { PORT, HOST,  } = app.config;
   try {
-    server.decorate('config', config);
-    // console.log(server.config
-    await server.listen({ port: 5000, host: '0.0.0.0' });
```
Comment (oh-math) [inline src/server.ts:15]: why are you pointing to a invalid IP in `host: '0.0.0.0' ` ?

File: src/server/index.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import Fastify from 'fastify';
+import { fastifyAutoload } from '@fastify/autoload';
+import { join } from 'node:path';
+import { config } from '@/config/env';
+import { prettyLog } from '@/utils/prettyLog';
+
+export async function buildServer() {
+  const server = Fastify({
+    logger: {
+      transport: prettyLog,
```
Comment (oh-math) [inline src/server/index.ts:10]: One doubt... This piece is creating a log struct with a configuration that prettify logs?

File: src/server/index.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  });
+
+  try {
+    // decorate instance with envVars
+    server.decorate('config', config);
+    server.register(fastifyAutoload, {
+      dir: join(__dirname, '../modules'),
+    });
+  } catch (e: unknown) {
+    server.log.error(e, 'error while mounting fastify')
```
Comment (oh-math) [inline src/server/index.ts:21]: better than "deu merda"

File: src/server/index.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    },
+  });
+
+  try {
+    // decorate instance with envVars
+    server.decorate('config', config);
+    server.register(fastifyAutoload, {
+      dir: join(__dirname, '../modules'),
+    });
+  } catch (e: unknown) {
```
Comment (oh-math) [inline src/server/index.ts:20]: what you think about use type guards? since you cannot type the error

File: src/server/index.ts (hunk 3 above)
Comment (Joabesv) [reply src/server/index.ts:10]: yep, if you see the `prettyLog` import it is the instance of a logger prettify config

File: src/server/index.ts (hunk 4 above)
Comment (Joabesv) [reply src/server/index.ts:21]: lol, forgot to remove that part
thx for the reminder

Comment (Joabesv) [reply src/server/index.ts:21]: fixed in recent commit

File: src/server/index.ts (hunk 5 above)
Comment (Joabesv) [reply src/server/index.ts:20]: plan to implement this later, in the current scope i can't think of type guard possibilites
will keep this as a reminder.

**Pull request merged**